### PR TITLE
Fix podman-remote run --attach stdin to show container ID

### LINKED ISSUE #15746: run --attach stdin failed to print container ID on podman-remote
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

**Steps to reproduce the issue:**

```
# podman-remote run --attach stdin quay.io/libpod/testimage:20220615
<No output>
```

**Describe the results you received:**

See above.

**Describe the results you expected:**

```
# podman-remote run --attach stdin quay.io/libpod/testimage:20220615
<container ID>
```

**Additional information you deem important (e.g. issue happens only occasionally):**

Succeeded to print container ID on local.
```
# podman run --attach stdin quay.io/libpod/testimage:20220615
272dea3eed717d7117a274d0c0f3467ebb89dc26550df356703143593816127c
```

**Output of `podman version`:**

<details>
<summary>podman-remote version</summary>

```
Client:       Podman Engine
Version:      4.3.0-dev
API Version:  4.3.0-dev
Go Version:   go1.18.3
Git Commit:   1635fe8620fa454647cf7db2af9ed0c9b52eacfb
Built:        Mon Sep 12 14:06:11 2022
OS/Arch:      linux/amd64

Server:       Podman Engine
Version:      4.3.0-dev
API Version:  4.3.0-dev
Go Version:   go1.18.3
Git Commit:   1635fe8620fa454647cf7db2af9ed0c9b52eacfb
Built:        Mon Sep 12 14:06:02 2022
OS/Arch:      linux/amd64
```
</details>

**Output of `podman info`:**

<details>
<summary>podman-remote info</summary>

```
host:
  arch: amd64
  buildahVersion: 1.28.0-dev
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - misc
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpuUtilization:
    idlePercent: 99.89
    systemPercent: 0.03
    userPercent: 0.08
  cpus: 12
  distribution:
    distribution: fedora
    variant: server
    version: "36"
  eventLogger: journald
  hostname: fedora36
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.18.11-200.fc36.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 1826119680
  memTotal: 8326582272
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.5-1.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.5
      commit: 54ebb8ca8bf7e6ddae2eb919f5b82d1d96863dea
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: unix:///run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 8314941440
  swapTotal: 8325689344
  uptime: 269h 1m 48.00s (Approximately 11.21 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /usr/share/containers/storage.conf
  containerStore:
    number: 1
    paused: 0
    running: 0
    stopped: 1
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 106285760512
  graphRootUsed: 25527275520
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.3.0-dev
  Built: 1662959162
  BuiltTime: Mon Sep 12 14:06:02 2022
  GitCommit: 1635fe8620fa454647cf7db2af9ed0c9b52eacfb
  GoVersion: go1.18.3
  Os: linux
  OsArch: linux/amd64
  Version: 4.3.0-dev
```
</details>

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**
KVM, fedora36

### PR DESCRIPTION
**Pull request by boaz0**
#### Does this PR introduce a user-facing change?

```release-note
Bugfix to podman-remote run --attach stdin does not show container ID
```

closes #15746

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
LGTM, thanks!

In future commits, please make sure to add a more descriptive explanation of the changes to the commit message.

@containers/podman-maintainers

File: test/system/030-run.bats [hunk 1]
```diff
@@ -892,4 +892,14 @@ $IMAGE--c_ok" \
     run_podman container rm -f -t 0 c_ok c_fail_no_rm
 }
 
+@test "podman run --attach stdin prints container ID" {
+    ctr_name="container-$(random_string 5)"
+    run_podman run --name $ctr_name --attach stdin $IMAGE echo hello 2>/dev/null
```
Comment (edsantiago) [inline test/system/030-run.bats:897]: The `2>/dev/null` here is a NOP, so it's misleading to include it.

File: test/system/030-run.bats [hunk 2]
```diff
@@ -892,4 +892,14 @@ $IMAGE--c_ok" \
...
+    run_output=$output
+    run_podman ps -a --filter name=$ctr_name --format "{{.ID}}"
+    ctr_id=$output
+    is "$run_output" ".*$ctr_id.*" "Did not find container ID in the output"
```
Comment (edsantiago) [inline test/system/030-run.bats:901]: We avoid wildcard matches whenever possible: it's too easy to miss warnings or other undesired output. Tight exact matches are best.

**Review by edsantiago — COMMENTED** [review]
Suggestion:
```diff
     ctr_name="container-$(random_string 5)"
-    run_podman run --name $ctr_name --attach stdin $IMAGE echo hello 2>/dev/null
+    run_podman run --name $ctr_name --attach stdin $IMAGE echo hello
     run_output=$output
-    run_podman ps -a --filter name=$ctr_name --format "{{.ID}}"
+    run_podman inspect --format "{{.Id}}" $ctr_name
     ctr_id=$output
-    is "$run_output" ".*$ctr_id.*" "Did not find container ID in the output"
+    is "$run_output" "$ctr_id" "Did not find container ID in the output"
     run_podman rm $ctr_name
```
...with explanations inline

**Comment by edsantiago** [comment]
Test LGTM. I confirmed that the new test fails under podman-remote @ main (as it should).

**Review by edsantiago — APPROVED** [review]

**Pull request merged**